camel/functions/retrieve_tools_function.py

```diff
@@ -0,0 +1,91 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from typing import List, Optional
+
+import numpy as np
+import torch
+from transformers import AutoModel, AutoTokenizer
+
+from camel.functions.openai_function import OpenAIFunction
+
+
+def embed_text(text: str, model: AutoModel, tokenizer: AutoTokenizer):
+    r"""
+    Embed text using chosen model and tokenizer.
+    """
+    inputs = tokenizer(text, return_tensors="pt")
+    with torch.no_grad():
+        outputs = model(**inputs)
+
+    return outputs.last_hidden_state.mean(dim=1).squeeze().cpu().numpy()
```

Comment:
Some model may not use the mean pooling / aggregation. For example this model uses last token pooling https://huggingface.co/intfloat/e5-mistral-7b-instruct Also, one frequent text embedding is just to use the CLS token. IMO either you can use existing text embeddings https://github.com/camel-ai/camel/blob/master/camel/embeddings/sentence_transformers_embeddings.py or implement this in the embeddings folder with class name such as TransformersEncoder with different pooling / aggregation strategy